File: OsmAnd-java/src/main/java/net/osmand/binary/StringBundleXmlReader.java
package net.osmand.binary;

import net.osmand.PlatformUtil;

import org.apache.commons.logging.Log;
import org.xmlpull.v1.XmlPullParser;

public class StringBundleXmlReader extends StringBundleReader {

	public final static Log log = PlatformUtil.getLog(StringBundleXmlReader.class);

	private XmlPullParser parser;

	public StringBundleXmlReader(XmlPullParser parser) {
		this.parser = parser;
	}

	@Override
	public void readBundle() {
		StringBundle bundle = getBundle();
		for (int i = 0; i < parser.getAttributeCount(); i++) {
			String name = parser.getAttributeName(i);
			String value = parser.getAttributeValue(i);
			bundle.putString(name, value);
		}
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/util/StringBundleXmlReader.kt
package net.osmand.shared.util

import net.osmand.shared.xml.XmlPullParser

class StringBundleXmlReader(private val parser: XmlPullParser) : StringBundleReader() {

	companion object {
		private val log = LoggerFactory.getLogger("StringBundleXmlReader")
	}

	override fun readBundle() {
		val bundle = getBundle()
		for (i in 0 until parser.getAttributeCount()) {
			val name = parser.getAttributeName(i)
			val value = parser.getAttributeValue(i)
			bundle.putString(name!!, value)
		}
	}
}
